glass/shells.py

```diff
@@ -470,36 +500,53 @@ def partition_lstsq(
     dz = np.gradient(zp)
 
     # create the window function matrix
-    a = [np.interp(zp, za, wa, left=0.0, right=0.0) for za, wa, _ in shells]  # type: ignore[arg-type]
-    a /= np.trapz(a, zp, axis=-1)[..., None]  # type: ignore[attr-defined]
+    a = np.array([np.interp(zp, za, wa, left=0.0, right=0.0) for za, wa, _ in shells])
+    a /= np.trapz(  # type: ignore[attr-defined]
+        a,
+        zp,
+        axis=-1,
+    )[..., None]
     a = a * dz
 
     # create the target vector of distribution values
-    b = ndinterp(zp, z, fz, left=0.0, right=0.0)  # type: ignore[no-untyped-call]
+    b = ndinterp(zp, z, fz, left=0.0, right=0.0)
     b = b * dz
 
     # append a constraint for the integral
     mult = 1 / sumtol
-    a = np.concatenate([a, mult * np.ones((len(shells), 1))], axis=-1)  # type: ignore[assignment]
-    b = np.concatenate([b, mult * np.reshape(np.trapz(fz, z), (*dims, 1))], axis=-1)  # type: ignore[attr-defined]
+    a = np.concatenate([a, mult * np.ones((len(shells), 1))], axis=-1)
+    b = np.concatenate(
+        [
+            b,
+            mult
+            * np.reshape(
+                np.trapz(  # type: ignore[attr-defined]
+                    fz,
+                    z,
+                ),
+                (*dims, 1),
+            ),
+        ],
+        axis=-1,
+    )
```

Comment:
What happened here?

Reply:
I was trying to make it clear that the `# type: ignore[attr-defined]` is specific to `np.trapz`. This is "fixed" in #391